### PR TITLE
Feature/pap 9 implement real time direct messaging dm functionality

### PR DESCRIPTION
**Pull request by bobbygrdn**
## Problem
Jira Ticket: https://project-ascend-io.atlassian.net/browse/PAP-9?atlOrigin=eyJpIjoiMTQzZDM5YTNmZWIxNDVlNmE1MjczNGI5YzM0N2FkYjkiLCJwIjoiaiJ9
Our application needed the ability to send direct messages from one user to another and a way to persist that data in the database.

## Solution
REST endpoints were created to handle the creation of messages and chats, which will be used to persist direct messages from one user to the other.

A Socket.io server was created to handle real-time events transmitted from one client to another, creating real-time direct messaging functionality.

Using AsyncAPI, documentation on how to interact with the Socket.io server from a client is generated using the generate:docs command. That documentation is delivered at the **'/socket-docs'** path.

## Type of change
- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?
Unit Tests
Manual Testing
Postman REST/Socket.io

## Checklist:
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by oestrada1001** [comment]
Great job @bobbygrdn! Code looks good. Let's keep this PR open until we can integrate it locally with the frontend to test it.

**Review by oestrada1001 — CHANGES_REQUESTED** [review]
Great job! I requested one minor improvement.

File: src/socket.ts [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    cors: {
+      origin: [`${process.env.CLIENT_SERVER}`, 'https://admin.socket.io'],
+      methods: ['GET', 'POST'],
+      credentials: true,
+    },
+  });
+
+  instrument(io, {
+    auth: false,
+    mode: 'development',
```
[the remark this answers is not in the document]
Comment (bobbygrdn) [reply src/socket.ts:26]: No. This is just a setup to add the AdminUI capabilities for development. We can add this to the production environment, but right now it serves as a debugging tool.

Here is the reference: https://socket.io/docs/v4/admin-ui/

[the remark this answers is not in the document]
Comment (oestrada1001) [reply src/socket.ts:26]: Great. That make sense for auth. Let's make `mode` the same value as NODE_ENV. Leveraging the .env will allow us to toggle the mode type based on environment without committing a new chance.

**Review by oestrada1001 — CHANGES_REQUESTED** [review]

File: src/socket.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (bobbygrdn) [reply src/socket.ts:26]: Sounds good. That will be an easy update. 😁

[the remark this answers is not in the document]
Comment (bobbygrdn) [reply src/socket.ts]: I have updated the code we discussed to reference the environment for mode and added authentication credentials, which will be retrieved from the environment file. Let me know what you think.

**Review by oestrada1001 — APPROVED** [review]
Love the abstraction! Great job @bobbygrdn 🙌

**Pull request merged**